From 6a2d48b2db2e5f0270fba7447e213b1b73fcc83e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 11 Sep 2020 23:40:58 +0300
Subject: [PATCH] [SW-17444] preboot: goya2_arc: enable security flow

This patch add support to check sys_cfg efuse values stored in
scratchpad SRAM and return true/false by check enable validation bit.

Change-Id: I3c6d0cc53330898716bba10c2fbf0a08517738a9
---

diff --git a/goya2_arc_gnu/security.c b/goya2_arc_gnu/security.c
index 4ef7d0c..65cc838 100644
--- a/goya2_arc_gnu/security.c
+++ b/goya2_arc_gnu/security.c
@@ -11,12 +11,22 @@
 #include <preboot/sha256.h>
 #include <goya2/goya2_security.h>
 
+#define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
+
 static u32 sp_sram_readl(u32 offset)
 {
 	u32 reg = *((volatile u32 *) (HL_SCRATCHPAD_SRAM_BASE + offset));
 	return reg;
 }
 
+bool preboot_security_is_enabled(void)
+{
+	u32 config_efuse;
+
+	config_efuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
+	return (config_efuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
+}
+
 bool preboot_security_key_is_valid(int key_idx, const void *pub_key,
 			 int pub_key_len, const void *pub_key_ext,
 			 int pub_key_ext_len)
